tachyon/crypto/challenger/challenger.h

```diff
@@ -111,7 +111,8 @@ class Challenger {
       return v != std::numeric_limits<uint32_t>::max();
     });
     CHECK(it != ret.end());
-    CheckWitness(bits, Field(*it));
+    Derived* derived = static_cast<Derived*>(this);
+    CHECK(derived->CheckWitness(bits, Field(*it)));
```

Comment:
Please add `[[nodiscard]]` to `CheckWitness()`, too!

```suggestion
    CHECK(CheckWitness(bits, Field(*it)));
```